tests/unit/auth/v4/generateV4Headers.js

```diff
@@ -0,0 +1,57 @@
+'use strict'; // eslint-disable-line strict
+
+const assert = require('assert');
+const http = require('http');
+
+const { generateV4Headers } = require('../../../../lib/auth/auth').client;
+
+const host = 'localhost:8000';
+const headers = {
+    host,
+};
+const token = 'token';
+const data = 'data';
+
+describe('v4 header generation', () => {
+    it('should add x-amz-security-token if needed', done => {
+        const req = http.request({ headers });
+
+        generateV4Headers(req, data, 'accessKey', 'secretKey', 'iam', null, token);
+
+        try {
+            assert.deepStrictEqual(req.getHeader('x-amz-security-token'), token);
+            return done();
+        } catch (err) {
+            return done(err);
+        }
```

Comment:
```suggestion
        assert.doesNotThrow(
            assert.deepStrictEqual(req.getHeader('x-amz-security-token'), token)
        );
```

Could use does not throw in place of a try catch, removes the need for done as well.

Reply:
I started without `done` but the test suite was hanging, not sure why as it should work, but settled on this.

Reply:
@miniscruff actually it just came back to me, `assert.doesNotThrow` will throw an `AssertionError` if the code to assert on throws an exception (be it `AssertionError` or anything else) so it would be back to square one.